.po template extraction fails due to unrecognized object type

happens after #8059 

https://github.com/cataclysmbn/Cataclysm-BN/actions/runs/23824865104/job/69445426078
```
---
File: 'data/json/recipes/nested_categories.json'
---
JSON error
ERROR: Unrecognized object type 'nested_category'!
--- JSON Item:
{'type': 'nested_category', 'id': 'nested_sandwiches', 'nested_name': 'sandwiches', 'category': 'CC_FOOD', 'subcategory': 'CSC_FOOD_NESTED', 'description': 'Assorted sandwiches.', 'nested_category_data': ['sandwich_sauce', 'sandwich_honey', 'sandwich_cheese', 'sandwich_cheese_grilled', 'sandwich_pb', 'sandwich_pbj', 'sandwich_pbh', 'sandwich_pbm', 'sandwich_fairy', 'sandwich_veggy', 'sandwich_jam', 'fish_sandwich', 'sandwich_deluxe', 'sandwich_t', 'sandwich_okay', 'sandwich_deluxe_nocheese']}
---
Error in extract_json_strings.py. Aborting
Error: Process completed with exit code 1.
```

solution: make the script skip not fail on unrecognized object type

--- a/lang/extract_json_strings.py
+++ b/lang/extract_json_strings.py
@@ -1011,6 +1011,7 @@ def extract_use_action_msgs(state, use_action, it_name):
             extract_use_action_msgs(state, v, it_name)
 
 found_types = set();
+warned_unknown_types = set();
 known_types = ignorable | extract_specials.keys() | automatically_convertible
 
 
@@ -1027,7 +1028,12 @@ def extract_json(state, item):
         extract_specials[object_type](state, item)
         return
     elif object_type not in automatically_convertible:
-        raise WrongJSONItem(f"ERROR: Unrecognized object type '{object_type}'!", item)
+        if object_type not in warned_unknown_types:
+            warned_unknown_types.add(object_type)
+            print(
+                f"WARNING: Skipping unrecognized object type '{object_type}' in '{state.current_source_file}'"
+            )
+        return
     if object_type not in known_types:
         print(f"WARNING: known_types does not contain object type '{object_type}'")
     # Use mod id as project name if project name is not specified

